lib/srv/db/endpoints.go

```diff
@@ -0,0 +1,72 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package db
+
+import (
+	"context"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/healthcheck"
+	"github.com/gravitational/teleport/lib/srv/db/mongodb"
+	"github.com/gravitational/teleport/lib/srv/db/mysql"
+	"github.com/gravitational/teleport/lib/srv/db/postgres"
+)
+
+type resolverClients interface {
+	mysql.ResolverClients
+}
+
+// checkSupportsHealthChecks returns nil if the database supports health checks.
+// TODO(gavin): add resolvers for all database protocols that we support, then
+// remove this helper.
```

Comment:
do we have plan to add the rest before release testing? if not, can we have an env var that enables a generic resolver and we can test it out during release testing?

i know redis requires some special parser. any other db protocol requires special handling?

how will this be added for Oracle since the lib is in e? (cc @Tener)